include/alpaka/kernel/TaskKernelCpuOmp2Blocks.hpp

```diff
@@ -38,12 +39,694 @@
+            //!
+            //! \tparam TLoopBody The loop body functor type.
+            //! \tparam TIdx The index type.
+            //!
+            //! \param kernel The kernel instance reference
+            //! \param loopBody The loop body functor instance, takes iteration index as input.
+            //! \param numIterations The number of loop iterations.
+            //! \param schedule The schedule object.
+            template<typename TLoopBody, typename TIdx>
+            ALPAKA_FN_HOST void operator()(
+                TKernel const& kernel,
+                TLoopBody&& loopBody,
+                TIdx const numIterations,
+                omp::Schedule const& schedule)
+            {
+                // Forward to ParallelForImpl that performs dispatch by by chunk size
+                switch(schedule.kind)
+                {
+                case omp::Schedule::NoSchedule:
+                    ParallelForImpl<TKernel, omp::Schedule, omp::Schedule::NoSchedule>{}(
+                        kernel,
+                        std::forward<TLoopBody>(loopBody),
+                        numIterations,
+                        schedule);
+                    break;
+                case omp::Schedule::Static:
+                    ParallelForImpl<TKernel, omp::Schedule, omp::Schedule::Static>{}(
+                        kernel,
+                        std::forward<TLoopBody>(loopBody),
+                        numIterations,
+                        schedule);
+                    break;
+                case omp::Schedule::Dynamic:
+                    ParallelForImpl<TKernel, omp::Schedule, omp::Schedule::Dynamic>{}(
+                        kernel,
+                        std::forward<TLoopBody>(loopBody),
+                        numIterations,
+                        schedule);
+                    break;
+                case omp::Schedule::Guided:
+                    ParallelForImpl<TKernel, omp::Schedule, omp::Schedule::Guided>{}(
+                        kernel,
+                        std::forward<TLoopBody>(loopBody),
+                        numIterations,
+                        schedule);
+                    break;
+#    if defined _OPENMP && _OPENMP >= 200805
+                case omp::Schedule::Auto:
+                    ParallelForImpl<TKernel, omp::Schedule, omp::Schedule::Auto>{}(
+                        kernel,
+                        std::forward<TLoopBody>(loopBody),
+                        numIterations,
+                        schedule);
+                    break;
+#    endif
+                case omp::Schedule::Runtime:
+                    ParallelForImpl<TKernel, omp::Schedule, omp::Schedule::Runtime>{}(
+                        kernel,
+                        std::forward<TLoopBody>(loopBody),
+                        numIterations,
+                        schedule);
+                    break;
+                    /// TODO: what do we do by default? Should not happen, but may due to copy-paste
```

Comment:
Please remove the comment. Compiler should find those cases when the kind member is an `enum class`

Reply:
It is not enum class, as explained at its definition